pkg/instancegroups/instancegroups.go

```diff
@@ -148,15 +149,40 @@ func (r *RollingUpdateInstanceGroup) RollingUpdate(rollingUpdateData *RollingUpd
 		}
 	}
 
-	for _, u := range update {
-		err = r.drainTerminateAndWait(u, rollingUpdateData, isBastion, sleepAfterTerminate)
+	settings := resolveSettings(cluster, r.CloudGroup.InstanceGroup, numInstances)
+
+	concurrency := 0
+	maxConcurrency := 1
+
+	if r.CloudGroup.InstanceGroup.Spec.Role == api.InstanceGroupRoleNode && !rollingUpdateData.Interactive {
+		maxConcurrency = settings.MaxUnavailable.IntValue()
+		if maxConcurrency == 0 {
```

Comment:
A subtlety of this placement is that masters, bastions, and interactive updates won't respect disabling of rolling updates. We might want to invert the sense of the enclosing if and have the body set `maxConcurrency = 1`